src/libraries/System.Private.CoreLib/src/System/Random.cs

```diff
@@ -182,6 +182,105 @@ public virtual void NextBytes(byte[] buffer)
+        /// </summary>
+        /// <param name="choices">The items to use to populate the array.</param>
+        /// <param name="length">The length of array to return.</param>
+        /// <typeparam name="T">The type of array.</typeparam>
+        /// <returns>An array populated with random items.</returns>
+        /// <exception cref="ArgumentException"><paramref name="choices" /> is empty.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="choices" /> is <see langword="null" />.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   <paramref name="length" /> is not zero or a positive number.
+        /// </exception>
+        /// <remarks>
+        ///   The method uses <see cref="Next(int)" /> to select items randomly from <paramref name="choices" />
+        ///   by index. This is used to populate a newly-created array.
+        /// </remarks>
+        public T[] GetItems<T>(T[] choices, int length)
+        {
+            ArgumentNullException.ThrowIfNull(choices);
+            return GetItems(new ReadOnlySpan<T>(choices), length);
+        }
+
+        /// <summary>
+        ///   Creates an array populated with items chosen at random from choices.
+        /// </summary>
+        /// <param name="choices">The items to use to populate the array.</param>
+        /// <param name="length">The length of array to return.</param>
+        /// <typeparam name="T">The type of array.</typeparam>
+        /// <returns>An array populated with random items.</returns>
+        /// <exception cref="ArgumentException"><paramref name="choices" /> is empty.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   <paramref name="length" /> is not zero or a positive number.
+        /// </exception>
+        /// <remarks>
+        ///   The method uses <see cref="Next(int)" /> to select items randomly from <paramref name="choices" />
+        ///   by index. This is used to populate a newly-created array.
+        /// </remarks>
+        public T[] GetItems<T>(ReadOnlySpan<T> choices, int length)
+        {
+            if (length < 0)
+                throw new ArgumentOutOfRangeException(nameof(length), SR.ArgumentOutOfRange_NeedNonNegNum);
+
+            T[] items = new T[length];
+            GetItems<T>(choices, items.AsSpan());
+            return items;
+        }
+
+        /// <summary>
+        ///   Performs an in-place shuffle of an array.
+        /// </summary>
+        /// <param name="values">The array to shuffle.</param>
+        /// <typeparam name="T">The type of array.</typeparam>
+        /// <exception cref="ArgumentNullException"><paramref name="values" /> is <see langword="null" />.</exception>
+        public void Shuffle<T>(T[] values)
+        {
+            ArgumentNullException.ThrowIfNull(values);
+            Shuffle<T>(values.AsSpan());
+        }
+
+        /// <summary>
+        ///   Performs an in-place shuffle of a span.
```

Comment:
Should the algorithm used be statet?
Maybe just under remarks and that it is an implementation detail subject to change.

Reply:
> Should the algorithm used be statet?

Why?  My default answer would be "no".

Reply:
There are several algorithms for shuffling, each has different characteristics. As a user of this API I want to know these characteristcs.
Similar to the note in [Random](https://learn.microsoft.com/en-us/dotnet/api/system.random?view=net-7.0#remarks) (aside that there's no note for the xoshiro256**).

Sure I can have a look into the code to see what's used, but for use case in simulation / numerical domain I don't expect that user will check the code, but they're interested in the type of algorithm being used.

Reply:
If someone needs a specific algorithm but we can change it at any time, how can they rely on it?

Reply:
If it's changed, then the docs should be updated too.

~The docs should state at least something like "Implementation similar to Fisher-Yates-Shuffle with O(n) time-complexity".~

For me the relevant points are:
* O(n)
* based on Random, so for reproducable results (e.g. testing, debuggin) `new Random(42)` can be used

Therefore I should be stated something like "Implementation uses System.Random internally and has O(n) time-complexity".
Then it's free to change the implementation (I think it won't be changed to something with higher time-complexity), and a piece of info about the algorithm is given.

Reply:
I have less concerns if you're talking about characteristics that we'd never change as opposed to naming specific algorithms.  Stating it makes an approximately linear number of calls deterministically to other methods on the Random instance would be fine.

Reply:
I pushed up some changes that indicate `Next(int, int)` is used and the time complexity.

Reply:
> less concerns if you're talking about characteristics that we'd never change as opposed to naming specific algorithms.

👍🏻 my thoughts evolved while we discussed that, so thanks therefore.